app/controllers/participating_sellers_controller.rb

```diff
@@ -0,0 +1,38 @@
+class ParticipatingSellersController < ApplicationController
+  before_action :set_participating_seller, only: %i[show update]
+
+  def index
+    json_response(ParticipatingSeller.all)
+  end
+
+  # POST /participating_sellers
+  def create
```

Comment:
We should generate and set the secret right here. You can generate a SHA the same way we do for our gift cards here

Reply:
I added it to the model like for sellers gift cards, does that work?

Reply:
yeah